tank/core/utils.py

```diff
@@ -44,3 +46,13 @@ def grep_dir(dirname: str, filter_regex: str = None, isdir: bool = False):
         contents = filter(lambda name: os.path.isdir(os.path.join(dirname, name)), contents)
 
     return contents
+
+
+def check_file_rights(file: str, mode: str):
+    file_stat: os.stat_result = os.stat(file)
+
+    # oct(file_stat.st_mode) returns '0o100XXX' <- last three numbers under X show file's permission mask
+    file_mode = oct(file_stat.st_mode)[-3:]
```

Comment:
You should operate not with strings but with numbers via https://docs.python.org/3/library/stat.html